osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -528,6 +528,19 @@ protected HitObjectComposer(Ruleset ruleset)
 
         public virtual string ConvertSelectionToString() => string.Empty;
 
+        /// <summary>
+        /// The custom logic that decides whether a HitObject should be selected when clicking an editor timestamp link
+        /// </summary>
+        /// <param name="hitObject">The hitObject being checked</param>
+        /// <param name="objectInfo">A single hitObject's information created with <see cref="ConvertSelectionToString"/></param>
+        /// <returns>Whether a HitObject should be selected or not</returns>
+        public virtual bool HandleHitObjectSelection(HitObject hitObject, string objectInfo) => false;
```

Comment:
This also looks off. Why does this method take in a `HitObject` _and_ some string? The `HitObject` should unambiguously indicate what to select.

Reply:
Refactored in [`4e1e197`](https://github.com/ppy/osu/pull/25371/commits/4e1e19728cb8ff9e11f18ab9c0e8635c2cc2ba9a)